modules/sso.php

```diff
@@ -823,11 +823,16 @@ function handle_login() {
 			}
 
 			if ( ! Jetpack::is_user_connected( $user->ID ) ) {
+				$calypso_env = ! empty( $_GET['calypso_env'] )
+					? $_GET['calypso_env']
```

Comment:
This should probably have some escaping of some sort applied to it, depending on what sorts of values can be passed here.

At least something like `esc_url_raw()`, if not also sanitizing the value through something much more restrictive (if possible).